src/utils/ThreadPool.h

```diff
@@ -0,0 +1,104 @@
+#include <condition_variable>
+#include <exception>
+#include <functional>
+#include <future>
+#include <mutex>
+#include <optional>
+#include <queue>
+#include <thread>
+#include <vector>
+
+#pragma once
+
```

Comment:
please add appropriate namespaces such as the others utils files
at least the "utils"
```
namespace UTILS
{
```